pipenv/core.py

```diff
@@ -1398,6 +1402,9 @@ def pip_install(
                 crayons.normal('pipenv lock', bold=True)
             )
         )
+        # Unlink possible tmp file created by us
+        if tmp_created:
+            os.unlink(tmp_created)
```

Comment:
Overall we can never guarantee that this handle will be closed, especially on windows, and we can't even guarantee that we will still have permission to delete it depending on the circumstances -- especially if there was an error of some kind during resolution during which time we are kicking off `subprocess` calls which, on windows, are known to leak open file handles

Reply:
OK. 

BTW the tests were not failing due to this change, at the time they were failing also without the change :). Second, this was tested on windows. Third, the tmp_created was set (and therefore unlink() attempted) only if the function was called with r == None (e.g. the file was created by this call to  pip_install() itself not if it was created outside the pip_install(). The r or the fd was never returned either directly or undirectly (as far as I could see) and the file was written / accessed via "with statement" (that change seemed to have been made bit earlier to guarantee that the fd does get closed), if the unlink() fails due to errors like file handle leakage issues in windows subprocess handling, I cannot comment, but I would imagine we've lost at that time anyways :)